### PR TITLE
Apply scrollbar-at-top solution to cancer type summary chart

### PR DESCRIPTION
**Pull request by jtquach1**
Fix cBioPortal/cbioportal#7945

## Describe changes proposed in this pull request:
- Hid scrollbar below summary chart and added scrollbar at top of chart

### To do:
- generalize scrollbar-at-top solution, maybe write an extendable class

### Before
top_scrollbar_before
### After
top_scrollbar_after

#### Links used
- [Cancer type summary chart](https://www.cbioportal.org/results/cancerTypesSummary?genetic_profile_ids_PROFILE_MUTATION_EXTENDED=laml_tcga_pan_can_atlas_2018_mutations&cancer_study_list=laml_tcga_pan_can_atlas_2018%2Cacc_tcga_pan_can_atlas_2018%2Cblca_tcga_pan_can_atlas_2018%2Clgg_tcga_pan_can_atlas_2018%2Cbrca_tcga_pan_can_atlas_2018%2Ccesc_tcga_pan_can_atlas_2018%2Cchol_tcga_pan_can_atlas_2018%2Ccoadread_tcga_pan_can_atlas_2018%2Cdlbc_tcga_pan_can_atlas_2018%2Cesca_tcga_pan_can_atlas_2018%2Cgbm_tcga_pan_can_atlas_2018%2Chnsc_tcga_pan_can_atlas_2018%2Ckich_tcga_pan_can_atlas_2018%2Ckirc_tcga_pan_can_atlas_2018%2Ckirp_tcga_pan_can_atlas_2018%2Clihc_tcga_pan_can_atlas_2018%2Cluad_tcga_pan_can_atlas_2018%2Clusc_tcga_pan_can_atlas_2018%2Cmeso_tcga_pan_can_atlas_2018%2Cov_tcga_pan_can_atlas_2018%2Cpaad_tcga_pan_can_atlas_2018%2Cpcpg_tcga_pan_can_atlas_2018%2Cprad_tcga_pan_can_atlas_2018%2Csarc_tcga_pan_can_atlas_2018%2Cskcm_tcga_pan_can_atlas_2018%2Cstad_tcga_pan_can_atlas_2018%2Ctgct_tcga_pan_can_atlas_2018%2Cthym_tcga_pan_can_atlas_2018%2Cthca_tcga_pan_can_atlas_2018%2Cucs_tcga_pan_can_atlas_2018%2Cucec_tcga_pan_can_atlas_2018%2Cuvm_tcga_pan_can_atlas_2018&Z_SCORE_THRESHOLD=2.0&RPPA_SCORE_THRESHOLD=2.0&data_priority=0&profileFilter=0&case_set_id=all&gene_list=KRAS%253A%2520MUT%2520%253D%2520G12D&geneset_list=%20&tab_index=tab_visualize&Action=Submit)
- [Plots tab (box chart)](https://www.cbioportal.org/results/plots?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=tmb_mskcc_2018&case_set_id=tmb_mskcc_2018_sequenced&data_priority=0&gene_list=BRCA1&geneset_list=%20&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=tmb_mskcc_2018_mutations&plots_coloring_selection=%7B%7D&plots_horz_selection=%7B%22selectedDataSourceOption%22%3A%22CANCER_TYPE_DETAILED%22%7D&plots_vert_selection=%7B%7D&profileFilter=0&tab_index=tab_visualize)

### CODE REVIEW

**Comment by jtquach1** [comment]
Thinking of writing a class that both CancerTypeSummary and PlotsTab can both extend from (since I adopted a lot of autobinded methods from PlotsTab and used the same top scrollbar logic)

**Comment by alisman** [comment]
@jtquach1 rather than use inheritance, it would be better i think to just make a component which we could wrap around the div (that is being scrolled).   If it turns out it needs a reference to a child element, there should be a way to grab it using jquery.  in other words, just get the first child of the wrapper $(scrollWrapperEl).children()[0].  Let me know if you want to discuss further.

**Comment by jtquach1** [comment]
> @jtquach1 rather than use inheritance, it would be better i think to just make a component which we could wrap around the div (that is being scrolled). If it turns out it needs a reference to a child element, there should be a way to grab it using jquery. in other words, just get the first child of the wrapper $(scrollWrapperEl).children()[0]. Let me know if you want to discuss further.

@alisman sure, I'd like to discuss further about using jquery to create wrapper component.

**Comment by jtquach1** [comment]
Updated before and after:
### Before
plots_tab_before

### After
plots_tab_after

**Comment by jtquach1** [comment]
# Updated before and after:
## Scatter plot (minus ghost scrollbar)
image
image

## Waterfall plot (minus ghost scrollbar)
image
image

## Cancer types summary bar chart
### Doesn't have "ghost" scrollbar occupying space at top for plot that isn't wide enough to scroll (same as before)
image
image

## Box plot (same as before)
image
image

**Review by Luke-Sikina — APPROVED** [review]

**Review by dippindots — APPROVED** [review]
@jtquach1 Great work! Looks good to me!

File: src/shared/components/plots/BoxScatterPlot.tsx [hunk 1]
```diff
@@ -844,11 +849,7 @@ export default class BoxScatterPlot<
                     role="img"
                     viewBox={`0 0 ${this.svgWidth} ${this.svgHeight}`}
                     onMouseMove={this.onMouseMove}
-                    ref={ref => {
-                        if (this.props.svgRef) {
-                            this.props.svgRef(ref);
-                        }
-                    }}
+                    ref={this.svgRef()}
```
Comment (alisman) [inline src/shared/components/plots/BoxScatterPlot.tsx:852]: it's kind of strange that this gets invoked.  usually you just pass a callback to ref

Comment (alisman) [reply src/shared/components/plots/BoxScatterPlot.tsx:852]: like, why don't you just feed it this.props.svgRef ?

**Pull request merged**